src/main/java/org/springframework/data/elasticsearch/repository/support/ElasticsearchStringValueToStringConverter.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.data.elasticsearch.repository.support;
+
+import org.springframework.core.convert.TypeDescriptor;
+import org.springframework.core.convert.converter.GenericConverter;
+import org.springframework.lang.Nullable;
+
+import java.util.Collections;
+import java.util.Set;
+import java.util.regex.Matcher;
+
+/**
+ * Values in elasticsearch query may contain quotations and should be escaped when converting.
+ * However, the converter should only be used in this situation, rather than common string to string conversions.
+ *
+ * @author Haibo Liu
+ */
+public class ElasticsearchStringValueToStringConverter implements GenericConverter {
+
+	@Override
+	public Set<ConvertiblePair> getConvertibleTypes() {
+		return Collections.singleton(new ConvertiblePair(String.class, String.class));
+	}
+
+	@Override
+	@Nullable
+	public Object convert(@Nullable Object source, TypeDescriptor sourceType, TypeDescriptor targetType) {
+		if (source == null) {
+			return "null";
```

Comment:
why not returning `null`?

Reply:
Assuming that this is the query:
```json
{
  "query": {
    "term": {
      "nickname": {
        "value": null
      }
    }
  }
}
```
Elasticsearch can't accept `null` as value and will produce:
```json
{
  "error" : {
    "root_cause" : [
      {
        "type" : "illegal_argument_exception",
        "reason" : "value cannot be null"
      }
    ],
    "type" : "illegal_argument_exception",
    "reason" : "value cannot be null"
  },
  "status" : 400
}
```
So there should be two choices for us:
1. convert `null` to string `"null"` or empty `""`, and this will avoid the error;
2. throw an invalid value error directly;

There is [already a similar way](https://docs.spring.io/spring-data/elasticsearch/reference/elasticsearch/repositories/elasticsearch-repository-queries.html#elasticsearch.query-methods.at-query) to insert query parameters into placeholder(`?0`, ...), and the actually implementation in `StringQueryUtil` just treat `null` string as `"null"`:

https://github.com/spring-projects/spring-data-elasticsearch/blob/8f745b19d14ca8af07cda4161781deda305b873d/src/main/java/org/springframework/data/elasticsearch/repository/support/StringQueryUtil.java#L85

so I obey the same implementation here.

However, I now think this isn't appropriate because a `nickname` value may really be string `"null"` and this'll cause the unexpected results.

**I prefer to raise an error for `null` string now.**

When it comes to collection, things can be different. I think:
1. a `null` collection: can be interpreted into `[]`
2. a collection with `null` value `["a", null]`: we can ignore the `null` value in the collection and just keep the non-null strings to produce a collection value like `["a"]`

Reply:
The default `ConversionSerivce` will handle the conversion for `null` source value, rather than a specific converter, so I put the null check into where the SpEL expression is evaluated.